Tasks/DeployAzureResourceGroup/operations/MachineGroupExtensionHelper.ts

```diff
@@ -0,0 +1,250 @@
+import computeManagementClient = require("./azure-rest/azure-arm-compute");
+import util = require("util");
+import tl = require("vsts-task-lib/task");
+import azure_utils = require("./AzureUtil");
+import deployAzureRG = require("../models/DeployAzureRG");
+import constants = require("./Constants");
+
+export class MachineGroupExtensionHelper {
+    private _taskParameters: deployAzureRG.AzureRGTaskParameters;
+    private _azureUtils: azure_utils.AzureUtil;
+    private _computeClient;
+
+    constructor(taskParameters: deployAzureRG.AzureRGTaskParameters) {
+        this._taskParameters = taskParameters;
+        this._computeClient = new computeManagementClient.ComputeManagementClient(this._taskParameters.credentials, this._taskParameters.subscriptionId);
+        this._azureUtils = new azure_utils.AzureUtil(this._taskParameters, this._computeClient);
+    }
+    public async installExtension() {
+        var operationParameters = new OperationParameters("installation");
+        var listOfVms = await this._azureUtils.getVMDetails();
+        var vmInstanceViewPromises: Promise<any>[] = [];
+        operationParameters.vmCount = listOfVms.length;
+        for (var vm of listOfVms) {
+            var vmName = vm["name"];
+            var resourceGroupName = this._taskParameters.resourceGroupName;
+            vmInstanceViewPromises.push(this.getVmInstanceView(resourceGroupName, vmName, { expand: 'instanceView' }, operationParameters));
```

Comment:
You can choose to perform all operations on a vm in another method which returns a promise.